infra/helper.py

```diff
@@ -419,6 +462,117 @@ def check_build(args):
   return exit_code
 
 
+def _get_fuzz_targets(project_name):
+  """Return names of fuzz targest build in the project's /out directory."""
+  command = [
+      'find',
+      _get_output_dir(project_name),
+      '-maxdepth',
+      '1',
+      '-type',
+      'f',
+      '-executable'
+  ]
+  output = subprocess.check_output(command)
+  return [os.path.basename(path) for path in output.split()]
+
+
+def _get_latest_corpus(project_name, fuzz_target, base_corpus_dir):
+  """Download the latest corpus for the given fuzz target."""
+  corpus_dir = os.path.join(base_corpus_dir, fuzz_target)
+  if not os.path.exists(corpus_dir):
+    os.makedirs(corpus_dir)
+
+  corpus_backup_url = CORPUS_BACKUP_URL_FORMAT.format(project_name=project_name,
+                                                      fuzz_target=fuzz_target)
+  command = [
+      'gsutil',
+      'ls',
+      corpus_backup_url
+  ]
+  output = subprocess.check_output(command).splitlines()
+  if output:
+    latest_backup_url = output[-1]
+    archive_path = corpus_dir + '.zip'
+    command = [
+        'gsutil',
+        'cp',
+        latest_backup_url,
+        archive_path
+    ]
+    subprocess.check_call(command)
+    command = [
+        'unzip',
+        '-q',
+        '-o',
+        archive_path,
+        '-d',
+        corpus_dir
+    ]
+    subprocess.check_call(command)
+    os.remove(archive_path)
+  else:
+    # Sync the working corpus copy if a minimized backup is not available.
+    corpus_url = CORPUS_URL_FORMAT.format(project_name=project_name,
+                                          fuzz_target=fuzz_target)
+    command = [
+        'gsutil',
+        '-m',
+        '-q',
+        'rsync',
+        '-R',
+        corpus_url,
+        corpus_dir
+    ]
+    subprocess.check_call(command)
+
+
+def download_corpus(project_name):
+  """Download most recent corpus from GCS for the given project."""
+  fuzz_targets = _get_fuzz_targets(project_name)
+  corpus_dir = _get_corpus_dir(project_name)
+  if not os.path.exists(corpus_dir):
+    os.makedirs(corpus_dir)
+
+  def _download_for_single_target(fuzz_target):
+    _get_latest_corpus(project_name, fuzz_target, corpus_dir)
+
+  print('Downloading corpus for %s' % project_name)
+  thread_pool = ThreadPool(multiprocessing.cpu_count())
+  thread_pool.map(_download_for_single_target, fuzz_targets)
+
+
+def profile(args):
+  """Generate code coverage using clang source based code coverage."""
+  if not _check_project_exists(args.project_name):
+    return 1
+  if not args.no_corpus_download:
+    download_corpus(args.project_name)
+
+  env = [
+      'FUZZING_ENGINE=libfuzzer',
+      'PROJECT=%s' % args.project_name,
+      'SANITIZER=profile'
+  ]
+
+  run_args = _env_to_docker_args(env) + [
+      '-v', '%s:/out' % _get_output_dir(args.project_name),
+      '-v', '%s:/corpus' % _get_corpus_dir(args.project_name),
+      '-p', '8008:8008',
```

Comment:
Not that it matters, but It would be great to make the port configurable via a command line argument. It would help avoid collisions or make it possible to run `helper.py` inside VMs where not all ports are forwarded to the host.

Reply:
Sure, done!